code/modules/mob/living/carbon/human/human_helpers.dm

```diff
@@ -113,6 +113,92 @@
 		return
 	return held_item.GetID()
 
+/**
+ * Used to fetch all the cash a vendor can access from the human
+ * See /mob/living/carbon/human/proc/get_cash_list() proc for the list of items and inventory slots it searches through
+ */
+/mob/living/carbon/human/proc/get_accessible_cash()
+	var/available_cash = 0
+	var/list/cash_list = get_cash_list()
+	if(!length(cash_list))
+		return 0
+	for(var/found_item in cash_list)
+		if(istype(found_item, /obj/item/holochip))
+			var/obj/item/holochip/chip_stack = found_item
+			available_cash += chip_stack.credits
+		if(istype(found_item, /obj/item/card/id))
+			var/obj/item/card/id/id_card = found_item
+			available_cash += id_card.registered_account.account_balance
+	return available_cash
+
+/**
+ * Used by vendors to see if the human can spend a certain amount of cash.
+ * Returns FALSE if they cannot, TRUE if they can and withdraws the cash.
+ * Arguments:
+ * * to_spend = how much cash needs to be deducted
+ */
+/mob/living/carbon/human/proc/spend_cash(to_spend)
+	if(!to_spend)
+		return FALSE
+	if(to_spend > get_accessible_cash()) //If we don't have enough money, early return
+		return FALSE
+	var/list/cash_list = get_cash_list()
+	if(!length(cash_list)) //Another check, just in case
+		return FALSE
+	for(var/obj/item/holochip/chip_stack in cash_list)//Holochips take priority over ID cards
+		if(chip_stack.credits >= to_spend)
+			chip_stack.spend(to_spend, TRUE)
+			return TRUE
+		else
+			var/temp_value_holder = to_spend
+			to_spend -= chip_stack.credits
+			chip_stack.spend(temp_value_holder, TRUE)
+	for(var/obj/item/card/id/id_card in cash_list)
+		var/temp_cash_holder = id_card.registered_account.account_balance
+		if(temp_cash_holder >= to_spend)
+			id_card.registered_account.adjust_money(-to_spend)
+			return TRUE
+		else
+			id_card.registered_account.adjust_money(-to_spend)
+			to_spend -= temp_cash_holder
+	return FALSE
+
+/**
+ * Used to find which 'cash holding items' the human has that are accessible by a vendor.
+ * Searches through both hands, ID and belt slots. Looks inside of wallets and PDAs.
+ * Returns a list consisting of IDs and chip stacks.
+ */
+/mob/living/carbon/human/proc/get_cash_list(list/search_through = null)
```

Comment:
I think this can be problem when it's used by monkey..? I am not sure

Reply:
How would it be? Monkeys are /mob/living/carbon/monkey, and the vendor code has several ishuman() checks